crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_type_union.rs

```diff
@@ -151,35 +201,29 @@ pub(crate) fn unnecessary_type_union<'a>(checker: &mut Checker, union: &'a Expr)
 
                     checker.generator().expr(&union)
                 }
-            } else {
-                let elts: Vec<Expr> = type_exprs.into_iter().cloned().collect();
-                let types = Expr::Subscript(ast::ExprSubscript {
-                    value: Box::new(Expr::Name(ast::ExprName {
-                        id: Name::new_static("type"),
-                        ctx: ExprContext::Load,
-                        range: TextRange::default(),
-                    })),
-                    slice: Box::new(pep_604_union(&elts)),
-                    ctx: ExprContext::Load,
-                    range: TextRange::default(),
-                });
-
-                if other_exprs.is_empty() {
-                    checker.generator().expr(&types)
-                } else {
-                    let elts: Vec<Expr> = std::iter::once(types)
-                        .chain(other_exprs.into_iter().cloned())
-                        .collect();
-                    checker.generator().expr(&pep_604_union(&elts))
-                }
-            };
+            }
+        };
 
-            diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
-                content,
-                union.range(),
-            )));
-        }
+        // Mark [`Fix`] as unsafe when comments are in range.
+        let applicability = if checker.comment_ranges().intersects(union.range()) {
```

Comment:
Hmm, I looked through all the `intersects` usages and I only found one exception where we set `Applicability::Unsafe` if the node contains comments. All other usages don't offer a fix instead. I think we should do the same here.

Could you take a look if we should use `has_comments` instead?

Reply:
`has_comments` is just wrapping `intersects` that includes leading and trailing content.

For type annotations I cannot come up with a case where the fix is changing the leading/trailing content:

```diff
   a: ( # comment 1
-    type[int] | type[float] ) = 3
+    type[int | float] ) = 
```

I'm not absolutely sure though.